backports/socketpair/__init__.py

```diff
@@ -42,7 +42,7 @@ def socketpair(family=socket.AF_INET, type=socket.SOCK_STREAM, proto=0):
                     try:
                         csock.connect((addr, port))
                     except socket.error as e:
-                        if e.errno != errno.WSAEWOULDBLOCK:
+                        if e.errno != errno.EWOULDBLOCK:
```

Comment:
If I may make a suggestion: if this is being changed, why not just check both?

    if not e.errno in (errno.EWOULDBLOCK, errno.WSAEWOULDBLOCK):

Each one individually has some corner-case downside:

1. `EWOULDBLOCK` alone might fail if someone builds Python from source for Windows without the relevant preprocessor macro defined (as per the discussion @lukebakken linked). Yes, *most* users of Python (especially on Windows) don't build Python from source, but some people do, and I'm not sure that we should be breaking them.

2. `WSAEWOULDBLOCK` is Windows-specific, which prevents this function from being invoked unmodified on a non-Windows runtime - which I would say is useful for testing changes, and might be otherwise useful to someone.

Reply:
Actually, following the discussion links, the Pika issue links to some Twisted code, and honestly, I think their solution is the cleanest:

    if sys.platform == 'win32':
        from errno import WSAEWOULDBLOCK as EWOULDBLOCK
    else:
        from errno import EWOULDBLOCK

..then they use `EWOULDBLOCK` directly within the code.

@mhils and @lukebakken What do you think of combining the change in this pull request with such a conditional import?

Reply:
Sounds good to me. Feel free to make the change on my branch of course!